src/lib/env-checks.js

```diff
@@ -89,8 +89,11 @@ function isContentDeployPreview(productSlug) {
 	return isDeployPreview() && isProxiedProduct(productSlug)
 }
 
-function isDeployPreview() {
-	return process.env.IS_CONTENT_PREVIEW
+function isDeployPreview(productSlug) {
+	const isProductSlugMatching =
+		!productSlug || productSlug === process.env.PREVIEW_FROM_REPO
```

Comment:
Adding support for an optional `productSlug` here, which let's us answer the question "is it a deploy preview for a specific product"